mlflow/pyspark/ml/__init__.py

```diff
@@ -79,11 +79,15 @@ def _get_warning_msg_for_skip_log_model(model):
 
 
 def _should_log_model(spark_model):
+    from pyspark.ml.base import Model
     # TODO: Handle PipelineModel/CrossValidatorModel/TrainValidationSplitModel
     class_name = _get_fully_qualified_class_name(spark_model)
     if class_name in _log_model_allowlist:
         if class_name == "pyspark.ml.classification.OneVsRestModel":
             return _should_log_model(spark_model.models[0])
+        elif class_name == 'pyspark.ml.pipeline.PipelineModel':
+            return all(_should_log_model(stage) for stage in spark_model.stages
```

Comment:
Are there any non-Model stages (that aren't instances of `pyspark.ml.base.Model`) that we should exclude due to size?

Reply:
No.
The pipelineModel will only include stages of Model or Transformer, if it is Transformer, it will only include some parameters and it will be small size.